### DISCUSSION TITLE
Build predictability when exporting to requirements.txt

### DISCUSSION THREAD

**Discussion by tfcace** [python-poetry/poetry#7504] [Q&A]
Hi! I have a question regarding something that puzzles me.
In hope to reduce the size of our Docker images, I was exploring the idea of using `poerty export` with `pip install --no-cache-dir`.

Given that some of our dependencies are private, we cannot use hashes AFAIK.
To that end, will we be losing the predicability of install from the lockfile (`poetry install`)?

Thanks!

**Comment by radoering** [comment]
Even without hashes you still have the locked versions in the exported requirements.txt. However, without hashes someone may replace a package and you will not notice.

**Comment by tfcace** [comment]
Any way to have private repositories if we have full control over them?

**Reply by radoering** [reply]
https://python-poetry.org/docs/repositories/ ?

**Reply by tfcace** [reply]
Autocorrect hell. "have" should have been "hash".
Any way to hash private repositories if we have full control over them?

**Reply by radoering** [reply]
Hashes will be used for packages from PyPI as well as for packages from private repositories.

**Reply by tfcace** [reply]
great, this was very helpful!
